clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -972,16 +975,28 @@ private Record nextFetchedRecord() {
         private List<ConsumerRecord<K, V>> fetchRecords(int maxRecords) {
             if (isFetched)
                 return Collections.emptyList();
+            if (nextInlineException != null) {
```

Comment:
@hachikuji Hmm... I get that you want to recover it. But I still don't like the previous behavior and I think that behavior out of the scope of this patch. I would appreciate it if you can fix that behavior in your patch rather than I doing it here. Thanks :)